backend/src/Controller/Admin/CaptainFinancialSystem/AdminCaptainFinancialSystemAccordingOnOrderController.php

```diff
@@ -0,0 +1,148 @@
+
+    public function __construct(SerializerInterface $serializer, AutoMapping $autoMapping, ValidatorInterface $validator, AdminCaptainFinancialSystemAccordingOnOrderService $adminCaptainFinancialSystemAccordingOnOrderService)
+    {
+        parent::__construct($serializer);
+        $this->autoMapping = $autoMapping;
+        $this->validator = $validator;
+        $this->adminCaptainFinancialSystemAccordingOnOrderService = $adminCaptainFinancialSystemAccordingOnOrderService;
+    }
+
+    /**
+     * admin:Captain's Financial System According On Order
+     * @Route("captainfinancialsystemaccordingonorderbyadmin", name="createCaptainFinancialSystemAccordingOnOrderByAdmin", methods={"POST"})
+     * @IsGranted("ROLE_ADMIN")
+     * @param Request $request
+     * @return JsonResponse
+     *
+     * @OA\Tag(name="Captain's Financial System According On Order")
+     *
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="new Captain's Financial System According On Order",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="integer", property="categoryName"),
+     *          @OA\Property(type="number", property="countKilometersFrom"),
+     *          @OA\Property(type="number", property="countKilometersTo"),
+     *          @OA\Property(type="number", property="amount"),
+     *          @OA\Property(type="number", property="bounce"),
+     *          @OA\Property(type="number", property="bounceCountOrdersInMonth"),
+     *      )
+     * )
+     *
+     * @OA\Response(
+     *      response=201,
+     *      description="Returns Captain's Financial System According On Order",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
+     *          @OA\Property(type="integer", property="id"),
+     *          @OA\Property(type="string", property="categoryName"),
+     *          @OA\Property(type="integer", property="countKilometersFrom"),
+     *          @OA\Property(type="integer", property="countKilometersTo"),
+     *          @OA\Property(type="number", property="amount"),
+     *          @OA\Property(type="number", property="bounce"),
+     *          @OA\Property(type="number", property="bounceCountOrdersInMonth"),
+     *      )
+     *   )
+     * )
+     * 
+     * @Security(name="Bearer")
+     */
+    public function createCaptainFinancialSystemAccordingOnOrder(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, AdminCaptainFinancialSystemAccordingOnOrderCreateRequest::class, (object)$data);
+
+        $violations = $this->validator->validate($request);
```

Comment:
FYI: there is no validator for AdminCaptainFinancialSystemAccordingOnOrderCreateRequest in the validation file.